core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,412 @@
+      'This subtopic is to test curriculum admin utility.'
+    );
+    await this.clickOn(subtopicPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createSubTopicButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(createSubTopicButton);
+    await this.page.waitForSelector(`${saveTopicButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(saveTopicButton);
+    await this.clickOn(closeSaveModalButton);
+  }
+
+  /**
+   * Function for creating a story for a certain topic.
+   */
+  async createStory(): Promise<void> {
+    await this.openTopicEditor();
+    await this.clickOn(addStoryButton);
+    await this.type(storyTitleField, 'Test Story 1');
+    await this.type(storyUrlFragmentField, 'test-story-one');
+    await this.type(
+      storyDescriptionField,
+      'This story is to test curriculum admin utility.'
+    );
+    await this.clickOn(storyPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createStoryButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(createStoryButton);
+  }
+
+  /**
+   * Function for creating a chapter for a certain story.
+   */
+  async createChapter(explorationId: string): Promise<void> {
+    await this.clickOn(addChapterButton);
+    await this.type(chapterTitleField, 'Test Chapter 1');
+    await this.type(chapterExplorationIdField, explorationId);
+    await this.clickOn(chapterPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createChapterButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(createChapterButton);
+  }
+
+  /**
+   * Function for publishing a story as a curriculum admin.
+   */
+  async publishStory(): Promise<void> {
+    await this.page.waitForSelector(`${saveStoryButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(saveStoryButton);
+    await this.type(
+      saveStoryMessageInput,
+      'Test publishing story as curriculum admin.'
+    );
+    await this.page.waitForSelector(`${closeSaveModalButton}:not([disabled])`);
+    await this.clickOn(closeSaveModalButton);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(publishStoryButton);
+  }
+
+  /**
+   * This function checks if the topic with given subtopic and skill is published.
+   */
+  async expectPublishedTopicToBePresent(): Promise<void> {
```

Comment:
How else would you recommend to verify the contents of the page? Does this.page.eval textContent work?

I'm not sure if this.page.eval itself is a problem since all other utility files follow the same pattern for expectance functions, in order to verify page contents.

Reply:
I think I gave a suggestion in my first paragraph? The thing I don't understand about the current logic is that it seems to be opening the topic editor and looking for a subtopic and skill name; none of this seems to have anything to do with whether the topic is published or not. (Or did I misunderstand the question in your first paragraph?)

For this.page.eval ... I guess that is OK if it can be mapped to the actions of an end user, e.g. look at the first topic in the list and verify that its name is XYZ. I think I was concerned that I couldn't do that with this function, but you're right that it might not be an issue with this.page.$eval itself.

Reply:
I'm not really sure what you initially meant by navigating around to see what the learner can see. 

Also I believe I misunderstood the functionality and the phrasing of my comments can be worked upon as well, the expectance function was meant to verify if the topic with the subtopic and skill is published -- and verify that the names of the subtopic and skill are correct, but I overlooked that topics can be saved and don't have to be published for them to exist.

I'll update the functions accordingly - thanks!

Reply:
> I'm not really sure what you initially meant by navigating around to see what the learner can see.

Ah sorry. If you want to check that a topic is published, then go to /learn/math or the relevant topic viewer page and see if some other random user can see it. (I think unpublished topics are not viewable from the learner side.)

But now that Im not quite sure what exactly you're trying to verify, the above might be moot :P